src/Tizen.NUI/src/internal/Common/ProcessorController.cs

```diff
@@ -0,0 +1,203 @@
+
+namespace Tizen.NUI
+{
+    /// <summary>
+    /// Singleton class.
+    /// ProcessorController handles ProcessorCallbacks those are requested to be called for each end of event process.
+    /// If there are works those required to be called after all of the event process for the efficiency, add them ProcessorController.
+    /// The calling order is not guaranteed but ProcessorCallbackes of LayoutController those are added by using AddLayoutControllerProcessor,
+    /// are called after ordinaly ProcessCallbacks added by AddProcessor.
+    ///
+    /// The added callbacks will be called every time of event tick.
+    /// If the callback is not be wanted to be called anymore, remove it.
+    /// </summary>
+    internal sealed class ProcessorController : Disposable
+    {
+        [UnmanagedFunctionPointer(CallingConvention.StdCall)]
+        internal delegate void ProcessorCallback();
+
+        private EventHandler processorOnceEventHandler;
+        private ProcessorOnceEventCallbackType processorOnceEventCallback;
+        private EventHandler processorEventHandler;
+        private ProcessorEventCallbackType processorEventCallback;
+        private EventHandler layoutProcessorEventHandler;
+        private LayoutProcessorEventCallbackType layoutProcessorEventCallback;
+
+        [UnmanagedFunctionPointer(CallingConvention.StdCall)]
+        private delegate void ProcessorOnceEventCallbackType();
+
+        [UnmanagedFunctionPointer(CallingConvention.StdCall)]
+        private delegate void ProcessorEventCallbackType();
+
+        [UnmanagedFunctionPointer(CallingConvention.StdCall)]
+        private delegate void LayoutProcessorEventCallbackType();
+
+        private ProcessorCallback callback = null;
+
+        public event EventHandler ProcessorOnceEvent
+        {
+            add
+            {
+                if (processorOnceEventHandler == null)
+                {
+                    processorOnceEventCallback = OnProcessorOnceEvent;
+                }
+
+                processorOnceEventHandler += value;
+            }
+
+            remove
+            {
+                processorOnceEventHandler -= value;
+
+                if (processorOnceEventHandler == null)
+                {
+                    processorOnceEventCallback = null;
+                }
+            }
+        }
+
+        public event EventHandler ProcessorEvent
+        {
+            add
+            {
+                if (processorEventHandler == null)
+                {
+                    processorEventCallback = OnProcessorEvent;
+                }
+
+                processorEventHandler += value;
+            }
+
+            remove
+            {
+                processorEventHandler -= value;
+
+                if (processorEventHandler == null)
+                {
+                    processorEventCallback = null;
+                }
+            }
+        }
+
+        public event EventHandler LayoutProcessorEvent
```

Comment:
no need to reimplement add/remove for simple event.

Reply:
Thank you I removed it!.